### DISCUSSION TITLE
用 web3 box 上传一个文件，需要 approve 9 次？！

### DISCUSSION THREAD

**Discussion by linghuccc** [IntensiveCoLearning/Web3-URL#162] [Q&A] [answered]
由于合约没有开源，不知道具体实现，不过根据 transaction hash，在我之前上传文件的人（ web3 box 合约创建者），也是 approve 了 9 次（但和我不同，他 approve 的 9 个 transaction 中间有 2 个其他 method）。

https://explorer.testnet.l2.quarkchain.io/address/0x42aD55767720A8df729EB161BA23249Dc821C7a9

**Comment by qzhodl** [comment, marked as answer]
是文件比较大吗？因为transaction携带calldata大小的限制，每次最多只能上传接近24KB大小的数据，如果文件超过这个大小，就需要把文件切片成多个chunk，分多个transaction上传。
合约的代码在这里：https://github.com/ethstorage/w3box/tree/main/contracts
